test/plugin/base_test.rb

```diff
@@ -1466,6 +1603,22 @@ def setup_container_metadata_stubs
                           'KUBE_BEARER_TOKEN: AoQiMuwkNP2BMT0S')
   end
 
+  def set_up_k8s_metadata_stubs
+    stub_metadata_request(
+      'instance/attributes/',
+      "attribute1\ncluster-name\ncluster-location\nlast_attribute")
+    stub_metadata_request('instance/attributes/cluster-location', K8S_LOCATION2)
+    stub_metadata_request('instance/attributes/cluster-name', K8S_CLUSTER_NAME)
+  end
+
+  def set_up_no_k8s_metadata_stubs
+    # Simulate an environment with no new k8s endpoints present.
+    stub_request(:get, %r{.*instance/attributes/cluster-location.*})
+      .to_raise(Errno::EHOSTUNREACH)
```

Comment:
Umm, shouldn't these be 404s instead? The GCE metadata service will be there, just maybe won't have those attributes.

Reply:
Note: the metadata agent uses a very small timeout to talk to the metadata server (1s). We were assured that this value would be sufficient. This was, in fact, true until about 1w ago. Hopefully it's a transient blip that will be fixed.
The logging agent uses a much longer timeout (the default, I think, is 60s), so I wouldn't worry about it.